### PR TITLE
social: Use navigator.share where supported

### LINKED ISSUE #42: Use navigator.share for native share sheet
## Problem

Right now we have a permalink for sharing each post in its footer. But it's not obvious how to share this link easily on mobile.

## Proposal

Use navigator.share, which is supported on mobile. We can trivially fall back to the existing behavior when unsupported.

## Prior Art

I just implemented this at the Blaede Family Recipes: https://blaede.family/recipes/banana-bread/

Docs: https://web.dev/web-share/

### PR DESCRIPTION
**Pull request by cassidyjames**
Fixes #42

### CODE REVIEW

File: _includes/social.html [hunk 1]
```diff
@@ -91,12 +91,31 @@ <h3>Share</h3>
...
 
+<script>
+function share() {
+  plausible('Share: Permalink');
+
+  if (navigator.share) {
+    navigator.share({
+      title: "{{ page.title }}",
+      text: "{{ page.description }}",
+      url: "{{ site.url }}{{ site.baseurl }}{{ page.url }}",
```
Comment (btkostner) [inline _includes/social.html:110]: I'd make this the same as the href attribute and use `prepend`. Just cause I like consistency

**Review by btkostner — DISMISSED** [review]

File: _includes/social.html (hunk 1 above)
Comment (cassidyjames) [reply _includes/social.html:110]:
```suggestion
      url: "{{ page.url | prepend: site.baseurl | prepend: site.url }}",
```

File: _includes/social.html [hunk 2]
```diff
@@ -91,12 +91,31 @@ <h3>Share</h3>
...
+    })
+    .then(() => console.log("Successful share"))
```
[the remark this answers is not in the document]
Comment (cassidyjames) [reply _includes/social.html:112]: Makes sense to remove this one, but I do think logging the unsupported case is useful. It's not an error, per-say, but it's the exception and falls back to the link itself.

```suggestion
```

File: _includes/social.html [hunk 3]
```diff
@@ -91,12 +91,31 @@ <h3>Share</h3>
...
+    .catch((error) => console.log("Error sharing", error));
+
+    return false;
+  } else {
+    console.log("Sharing not supported");
```
Comment (cassidyjames) [inline _includes/social.html:117]:
```suggestion
    console.log("Sharing not supported; falling back to link");
```

**Review by btkostner — APPROVED** [review]

**Pull request merged**